mission_control/navigator_missions/nav_missions/detect_deliver.py

```diff
@@ -6,35 +6,28 @@
 import tf
 import tf.transformations as trns
 from navigator_msgs.msg import ShooterDoAction, ShooterDoActionGoal
-from navigator_msgs.srv import ObjectDBSingleQuery, ObjectDBSingleQueryRequest, CameraToLidarTransform, CameraToLidarTransformRequest
+from navigator_msgs.srv import CameraToLidarTransform,CameraToLidarTransformRequest
 from geometry_msgs.msg import Point
+from std_srvs.srv import SetBool, SetBoolRequest
 from twisted.internet import defer
 from image_geometry import PinholeCameraModel
 import navigator_tools
 
 
 class DetectDeliverMission:
     # Note, this will be changed when the shooter switches to actionlib
-    center_error_threshold = 0.1
-    width_proportion = 0.15  # Desired proportion of frame width symbol is present in
-    width_error_threshold = 0.03
     resp = None
 
     def __init__(self, navigator):
-        # tf_listener = navigator.tf.TransformListener(navigator.nh,
-        # history_length=genpy.Duration(1)) # short history length so that we
-        # cover history being truncated
-        self.center_error = 1
-        self.width_error = 1
         self.navigator = navigator
-        self.getShooterWaypoint = navigator.nh.get_service_client(
-            "/database/single", ObjectDBSingleQuery)
         self.cameraLidarTransformer = navigator.nh.get_service_client(
             "/camera_lidar_transformer/transform_camera", CameraToLidarTransform)
         self.shooterLoad = txros.action.ActionClient(
             self.navigator.nh, '/shooter/load', ShooterDoAction)
         self.shooterFire = txros.action.ActionClient(
             self.navigator.nh, '/shooter/fire', ShooterDoAction)
+        self.visionActivation = navigator.nh.get_service_client('/dock_shapes/run',SetBool)
```

Comment:
Small gripe, but you should use the vision proxy activation call for this (which would just mean changing your service host to `/get_shape/switch` and doing: `yield self.navigator.vision_proxies['dock_shapes'].start()/stop()` and would save a couple lines).

Or even cleaner:
```python
vision_proxy = self.navigator.vision_proxies['dock_shapes']
yield vision_proxy.start()
...
yield vision_proxy.stop()
```